library/Network/Wai/Cli.hs

```diff
@@ -56,23 +64,22 @@ instance Options WaiOptions where
     <*> simpleOption "graceful"          "serve-normally"    "Graceful shutdown mode. One of: none, serve-normally, serve-503"
     <*> simpleOption "devlogging"        Nothing             "Whether development logging should be enabled"
     where
-      availableProtocols = intercalate ", " $ concat
-        [coreProtocols, tlsProtocols, fcgiProtocol]
-      coreProtocols = ["http", "unix", "activate", "cgi"]
-      tlsProtocols =
-#ifdef WaiCliTLS
-        ["http+tls", "unix+tls", "activate+tls"]
-#else
-        []
+      availableProtocols = intercalate ", " $ (allProtocols \\ exclusions)
```

Comment:
I restructured this a bit - it was going to be messy with the interplay between WaiCliTLS and UnixSockets otherwise.

Reply:
Hmmmm the old way looks silly indeed. Why did I even do the concatenation thing. Facepalm.

I'm not sure we need to calculate exclusions at runtime either. Maybe GHC is smart enough to optimize this, but I'd still prefer the more verbose preprocessor-happy way:

```haskell
availableProtocols = intercalate ", " $ concat [
  "http", "cgi"
#ifdef WaiCliUnix
 , "unix", "activate"
#endif
#ifdef WaiCliTLS
 , "http+tls"
#endif
#if defined(WaiCliTLS) && defined(WaiCliUnix)
 , "unix+tls", "activate+tls"
#endif
 -- also fastcgi and whatever
]
```

Reply:
done.